java-tools/OsmAndServer/src/main/java/net/osmand/server/api/services/UserdataService.java

```diff
@@ -211,8 +230,42 @@ public ResponseEntity<String> uploadMultipartFile(MultipartFile file, PremiumUse
                 throw new OsmAndPublicApiException(ERROR_CODE_GZIP_ONLY_SUPPORTED_UPLOAD, "File is submitted not in gzip format");
 			}
 		}
+        error = validateFreeAccountFile(pu, zipfile);
+        if (error != null) {
+            return error;
+        }
+        
 		return uploadFile(zipfile, dev, name, type, clienttime);
 	}
+    
+    private ResponseEntity<String> validateFreeAccount(PremiumUsersRepository.PremiumUser pu, String type) {
+        if (pu.orderid == null) {
+            Iterable<UserFile> files = filesRepository.findAllByUserid(pu.id);
+            if (IterableUtils.size(files) > MAX_NUMBER_OF_FILES_FREE_ACCOUNT) {
+                return ResponseEntity.badRequest().body(String.format("File limit reached (%d)!", MAX_NUMBER_OF_FILES_FREE_ACCOUNT));
+            }
+            if (!type.equals(FILE_TYPE_FAVOURITES) && !type.equals(FILE_TYPE_SETTINGS)) {
+                return ResponseEntity.badRequest().body(String.format("Free account can upload files with type %s and %s, this file type is %s!",
+                        FILE_TYPE_FAVOURITES, FILE_TYPE_SETTINGS, type));
+            }
+        }
+        return null;
+    }
+    
+    private ResponseEntity<String> validateFreeAccountFile(PremiumUsersRepository.PremiumUser pu, InternalZipFile zipfile) {
```

Comment:
This method should be combined with public void validateUser(PremiumUsersRepository.PremiumUser user) { cause it has contrary checks